nav.go

```diff
@@ -457,7 +459,28 @@ func (nav *nav) position() {
 	}
 }
 
-func (nav *nav) preview(path string, win *win) {
+func (nav *nav) previewLoop(ui *ui) {
+	var path string
+	for {
+		p, ok := <-nav.previewChan
+		if !ok {
+			return
+		}
+		if len(p) != 0 {
+			win := ui.wins[len(ui.wins)-1]
+			nav.previewGen(p, win)
+			path = p
+		} else if len(gOpts.cleaner) != 0 && len(gOpts.cleaner) != 0 && nav.volatilePreview {
+			cmd := exec.Command(gOpts.cleaner, path)
+			if err := cmd.Run(); err != nil {
+				log.Printf("cleaning preview: %s", err)
+			}
+			nav.volatilePreview = false
+		}
+	}
+}
+
+func (nav *nav) previewGen(path string, win *win) {
```

Comment:
If you decide to get rid of the single line functions, maybe you can rename this back to `preview`.

Reply:
Right, done.